src/util/defs.h

```diff
@@ -3,26 +3,39 @@
 #include <Qt>
 
 // Maximum buffer length to each EngineObject::process call.
-//TODO: Replace this with mixxx::AudioParameters::bufferSize()
-constexpr unsigned int MAX_BUFFER_LEN = 160000;
```

Comment:
This buffer is also used for the scalers. I am a bit concerned if the reduction of buffer size has bad effects there. 
So I think it would be more safe to introduce another macro and replace every usage one by one. 
Maybe we can finally get rid of this fixed number at all.

Reply:
I've run the app and the test with ASAN, no issues so far.

Reply:
Did you use a 192 kHz audio file with extreme low pitch, rubberband and soutdtouch and keylock disabled enabled? 
As far as I know ASAN is able to detect some out of bound accesses but now ASAN findings is no proof that there are no.
An additional theoretic consideration is required IMHO.

Reply:
You're describing an extreme scenario.
I think what the mixxx team should do is:
- take the PR
- add bounds checking assertions in the place you're specially worried about, and even if possible make the buffer allocation use the exact amount or required space
- add unit tests to trigger the worst situation
- run the unit tests with assertions and asan

But if you don't do that then you have no idea anyway.

Reply:
Yes that is the sore point here and the reason why I have suggested to keep the high buffer in places where we have "no idea".
Than we can barrow them one by one with the method you suggested. 

Keeping this PR open as it is will likely lead to one more stalled PR in our list.

Reply:
The real question is will someone continue this work of tidying the audio buffers?
If yes then you can merge and continue on this, there is time until 2.5.
Otherwise we can simply close and not merge.

Reply:
>The real question is will someone continue this work of tidying the audio buffers?

We can't commit on that, because we are all volunteers. 

Closing this would be a waste of resource (developer and review), because this PR contains some very welcome changes. 

In Mixxx we normally do not commit work in progress changes. The main branch should still be usable without extra risk. 
So I like to merge all parts that do not involve a risk.

Reply:
I'll do one last change later so it'll be in a state you're more comfortable with.